### ISSUE TITLE
New feature: render an action in a view.

### ISSUE THREAD

**Issue by Locustv2** [yiisoft/yii2#10932]
Here is the problem (or scenario if you want):

We have an `Item` record which have relations such as `reviews` and `photos`.

If we are going to use `modules` to implement the different actions that we can do to `items`,  it would be like this:

```
modules
    items
        controllers
            - DefaultController
            - ReviewController
            - PhotoController
       views
            default
            review
            photo
```

Each controller has it's own action and view files. The `default` views are general details about the item. But now if i want to add small snippets of a review or photos in the `default` views:

* i should create a file in the default folder for it or add the html directly to my view file that i want it to appear. This can cause duplication of codes if i need the same on in other view files.
* i could create a file `review/snippet.php` then use `$this->render('/review/snippet', [/*params*/])` inside the view filed in the `default` folder. But still this doesn't make sense as if i modify the `snippet` file, i need to go around and do necessary modification (to parameters)

If we had some `renderAction($action)` in the view, which would return an action without applying the layout.

Example:
`$this->renderAction(['/items/review/snippet', 'id' => $model->id])`

This would return the action `/items/review/snippet` without any layout.

I think this would be of great help when creating reusable snippets.

**Comment by SilverFire** [comment]
I had the same idea some time ago and tried to implement something similar, but faced a bunch of problems:

1. The action which is called from the view must render the HTML without layout, like `renderPartial` usually do. I didn't find a proper way to flag this case for the action, therefore I failed to overcome this problem.
2. How about pre/post action triggers, cache, etc? The will be triggered and can modify the output and make the result unusable in the view
3. Performance. Using this approach instantly brings you redundant checks and DB queries, that slow down the entire application
4. etc. Don't remember the exact points.

Thus, I've decided to reject this idea in my project and don't want to bring it to the core.

I'm closing this feature request, at least untill the questions above will be covered with a good solution.
